src/torznab.ts

```diff
@@ -522,6 +522,20 @@ async function updateCaps(
 	}
 }
 
+export async function refreshIndexerCaps(): Promise<void> {
+	updateCaps(await getEnabledIndexers()); // Keeps valid indexers up to date
+	const indexersWithoutCaps = (await getAllIndexers()).filter(
+		(indexer) =>
+			indexer.searchCap === null ||
+			indexer.tvSearchCap === null ||
+			indexer.movieSearchCap === null ||
+			indexer.tvIdCaps === null ||
+			indexer.movieIdCaps === null ||
+			indexer.categories === null,
+	);
+	updateCaps(indexersWithoutCaps); // Will display errors if any
+}
+
```

Comment:
why do these separately?

Reply:
I'm curious if there is ever a situation where it wouldn't be an "all or nothing" - missing just one seems unlikely?

Reply:
`updateCaps(indexersWithoutCaps)` is what will fix the null caps once the indexer becomes available. This will log an error message for each indexer that can't be reached every time it's ran, which I feel is desirable.

`updateCaps(await getEnabledIndexers())` checks for the unlikely scenario that caps have been updated. No error message will ever come from this unless the indexer just failed.

If `getAllIndexers()` were to be used instead of splitting them up, any rate limited indexers would display the error, even if the caps aren't null. Which I feel is misleading.